src/api/FactoryAPI.ts

```diff
@@ -126,27 +166,21 @@ export class FactoryAPI {
       this.assert.common.greaterThanZero(unit, coreAPIErrors.QUANTITY_NEEDS_TO_BE_POSITIVE(unit));
     });
 
-    let minDecimals = new BigNumber(18);
-    let tokenDecimals;
     await Promise.all(
       components.map(async componentAddress => {
         this.assert.common.isValidString(componentAddress, coreAPIErrors.STRING_CANNOT_BE_EMPTY('component'));
         this.assert.schema.isValidAddress('componentAddress', componentAddress);
 
-        const tokenContract = await StandardTokenMockContract.at(componentAddress, this.web3, {});
-        try {
-          tokenDecimals = await tokenContract.decimals.callAsync();
-          if (tokenDecimals.lt(minDecimals)) {
-            minDecimals = tokenDecimals;
-          }
-        } catch (err) {
-          minDecimals = ZERO;
-        }
-
         await this.assert.erc20.implementsERC20(componentAddress);
       }),
     );
 
-    this.assert.core.validateNaturalUnit(naturalUnit, minDecimals, coreAPIErrors.INVALID_NATURAL_UNIT());
+    const minNaturalUnit = await this.calculateMinimumNaturalUnit(components);
+
+    this.assert.core.validateNaturalUnit(
```

Comment:
Why create a new `validateNaturalUnit` when we can use a common assertion `isGreaterOrEqualThan`?

Reply:
we dont need to, we can get rid of this validateNaturalUnit which was doing something else before. good call